File: tests/fixtures/vulnerable/js/sql-injection.js
// Test fixture: SQL Injection
// Should trigger: sql-injection

const express = require('express');
const app = express();

app.get('/user', (req, res) => {
  const query = `SELECT * FROM users WHERE id = ${req.query.id}`;
  db.query(query, (err, results) => {
    res.json(results);
  });
});

app.get('/search', (req, res) => {
  const searchQuery = `SELECT * FROM products WHERE name LIKE '%${req.query.term}%'`;
  db.execute(searchQuery, (err, results) => {
    res.json(results);
  });
});

module.exports = app;

File: tests/fixtures/vulnerable/py/sql-injection.py
# Test fixture: SQL injection
# Should trigger: sql-injection

from flask import Flask, request
import sqlite3

app = Flask(__name__)

@app.route('/user/<user_id>')
def get_user(user_id):
    query = f"SELECT * FROM users WHERE id = {user_id}"
    return db.execute(query)

@app.route('/search')
def search():
    term = request.args.get('term')
    query = f"SELECT * FROM products WHERE name LIKE '%{term}%'"
    return db.execute(query)
